mapiproxy/modules/mpm_cache.c

```diff
@@ -386,14 +387,14 @@ static NTSTATUS cache_pull_OpenMessage(struct dcesrv_call_state *dce_call,
 
 	message = talloc((TALLOC_CTX *)mpm, struct mpm_message);
 	NT_STATUS_HAVE_NO_MEMORY(message);
-	
-	message->session = mpm_session_init((TALLOC_CTX *)mpm, dce_call);
+
+	message->session = mpm_session_init(dce_call, NULL);
 	NT_STATUS_HAVE_NO_MEMORY(message->session);
```

Comment:
You are correct, there is indeed a check (btw, it does not respect the coding policy). 

However this is prone to errors and it can still causes crashes elsewhere such as in mpm_session_find_by_uuid: https://github.com/openchange/openchange/pull/389/files#diff-1acfc358c59edcd26a813334c163524cR203

In this function the session->uuid is not checked beforehand and will lead to a crash in GUID_equal.

If uuid is a required parameter, then it needs to be handled as such: add sanity checks on parameters, return an error if it is missing and propagate session creation failure to upper layers.